### PR TITLE
Default SchemaTransform configs to snake_case

### PR DESCRIPTION
**Pull request by ahmedabu98**
Bringing back the changes reverted in #31109

More details in #31061 and #31353

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Fixes #31353

**Comment by lostluck** [comment]
Probably does affect Go, but IIRC it just means non idiomatic go field names in generated structs. The existing schema handling code should handle making sure the field is Exported (upper case first character) with a manual field name override of the given snake case name.

**Comment by ahmedabu98** [comment]
Thanks for the insight @lostluck! From a cursory search I see [xlang/bigtableio/bigtableio.go](https://github.com/apache/beam/blob/e764fc9c17dbc575d4a16b5e6454f4d38ef752be/sdks/go/pkg/beam/io/xlang/bigtableio/bigtable.go#L102) is currently the only SchemaTransform use-case in the Go SDK, is that right?

**Comment by ahmedabu98** [comment]
R: @lostluck 

> manual field name override of the given snake case name.

let me know if the last commit does what you meant here

**Comment by lostluck** [comment]
> Thanks for the insight @lostluck! From a cursory search I see [xlang/bigtableio/bigtableio.go](https://github.com/apache/beam/blob/e764fc9c17dbc575d4a16b5e6454f4d38ef752be/sdks/go/pkg/beam/io/xlang/bigtableio/bigtable.go#L102) is currently the only SchemaTransform use-case in the Go SDK, is that right?

I have no idea. I do not keep track of cross language implementations and details about specific IOs. Your search would be as good as mine.

File: sdks/python/apache_beam/yaml/standard_io.yaml
Comment (Polber) [inline sdks/python/apache_beam/yaml/standard_io.yaml:1]: Also need modifications in https://github.com/apache/beam/blob/master/sdks/python/apache_beam/yaml/standard_providers.yaml

File: sdks/python/apache_beam/yaml/standard_io.yaml [hunk 1]
```diff
@@ -30,14 +30,14 @@
...
       'WriteToBigQuery':
         table: 'table'
-        create_disposition: 'createDisposition'
-        write_disposition: 'writeDisposition'
-        error_handling: 'errorHandling'
+        create_disposition: 'create_disposition'
+        write_disposition: 'write_disposition'
+        error_handling: 'error_handling'
         # TODO(https://github.com/apache/beam/issues/30058): Required until autosharding support is fixed
         num_streams: 'numStreams'
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/standard_io.yaml:42]:
```suggestion
        num_streams: 'num_streams'
```

File: sdks/python/apache_beam/yaml/standard_io.yaml [hunk 2]
```diff
@@ -206,25 +206,25 @@
     mappings:
       'ReadFromJdbc':
         driver_class_name: 'driverClassName'
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/standard_io.yaml:208]:
```suggestion
        driver_class_name: 'driver_class_name'
```

**Review by Polber — CHANGES_REQUESTED** [review]
Left some comments

**Comment by ahmedabu98** [comment]
Thanks @Polber for catching those, can you PTAL again?

**Comment by Polber** [comment]
@ahmedabu98 LGTM, but there is an open PR #31455 that would require a couple changes as well if it gets in before this PR is merged

**Review by Polber — APPROVED** [review]

**Comment by robertwb** [comment]
@chamikaramj would this impact using the transform service cross-version?

**Comment by chamikaramj** [comment]
So transform service just wrap expansion services. So change of behavior should be similar to the change of behavior of the Java expansion service.

Regarding, cross-version usage, I think what could break is use-cases that assume a schema from a old Beam version while trying to use the new Beam version for expansion. For example, hand-coded wrappers in customer's side. We are updating the affected Beam provided wrappers so I think the chances of someone running into issues due to this should be rare.

**Pull request merged**